packages/sit-onyx/src/components/OnyxIconButton/OnyxIconButton.vue

```diff
@@ -0,0 +1,112 @@
+<script lang="ts" setup>
+import OnyxIcon from "../OnyxIcon/OnyxIcon.vue";
+import OnyxLoadingIndicator from "../OnyxLoadingIndicator/OnyxLoadingIndicator.vue";
+import type { OnyxIconButtonProps } from "./types";
+
+const props = withDefaults(defineProps<OnyxIconButtonProps>(), {
+  disabled: false,
+  type: "button",
+  variation: "primary",
+});
+
+defineSlots<{
+  /** Slot for an custom icon */
+  default(props: Record<string, never>): unknown;
+}>();
+
+const emit = defineEmits<{
+  /** Emitted when the button is clicked (and is not disabled). */
+  click: [];
+}>();
+</script>
+
+<template>
+  <button
+    class="onyx-icon-button"
+    :aria-label="props.label"
+    :class="{
+      [`onyx-icon-button--${props.variation}`]: true,
+      'onyx-icon-button--loading': props.loading,
+    }"
+    :disabled="props.disabled || props.loading"
+    @click="emit('click')"
+  >
+    <OnyxLoadingIndicator v-if="props.loading" type="circle" />
+    <OnyxIcon v-else-if="props.icon" :icon="props.icon" size="24px" />
```

Comment:
```suggestion
    <OnyxIcon v-else-if="props.icon" :icon="props.icon" />
```

24px is the default size. Should we add a property to set it? We could allow the same icon sizes as the regular OynxIcon

Reply:
done